### ISSUE TITLE
fix(operators): `advection_scalar` is not dealiased while `jacobian` is; every simulation notebook uses it as the RHS

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#90]
Part of #80.

## Summary

`SpectralDerivative2D.jacobian(psi, q)` truncates the product with the grid's 2/3 mask before returning. `SpectralDerivative2D.advection_scalar(u, v, q)` (and the 3D version) computes `u·∇q` and returns the raw pointwise product without truncation. The two are mathematically the same term when `u, v = -ψ_y, ψ_x`, but they differ by O(1) on band-limited fields, and `advection_scalar` is what the notebooks, scripts and four `test_correctness.py` tests use as the nonlinear term of Navier–Stokes and QG.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, jax.random as jr
from spectraldiffx import FourierGrid2D, SpectralDerivative2D

N = 48
g = FourierGrid2D.from_N_L(Nx=N, Ny=N, Lx=2 * jnp.pi, Ly=2 * jnp.pi, dealias="2/3")
d = SpectralDerivative2D(grid=g)
key = jr.key(0)
mk = lambda k: d.apply_dealias(jr.normal(k, (N, N)))   # band-limited random fields
psi, q = mk(jr.fold_in(key, 0)), mk(jr.fold_in(key, 1))
u, v = d.velocity_from_streamfunction(psi)
adv = d.advection_scalar(u, v, q)
jac = d.jacobian(psi, q)
mask = g.dealias_filter()
frac = lambda f: float((jnp.abs(g.transform(f))**2 * (1 - mask)).sum() / (jnp.abs(g.transform(f))**2).sum())
print("energy outside mask: adv", frac(adv), "jac", frac(jac))
print("max|adv - jac| =", float(jnp.abs(adv - jac).max()))
print("max|apply_dealias(adv) - jac| =", float(jnp.abs(d.apply_dealias(adv) - jac).max()))
```

```
energy outside mask: adv 6.87e-01   jac 3.7e-32
max|adv - jac| = 208.5
max|apply_dealias(adv) - jac| = 0.0
```

## Root cause

- `spectraldiffx/_src/fourier/operators.py:380-396` (2D `advection_scalar`) and `:581-593` (3D): return `u * dq_dx + v * dq_dy` directly.
- `operators.py:314-333` (`jacobian`): ends with `self.apply_dealias(...)`.

Call sites that inherit the aliasing: `docs/notebooks/navier_stokes_2d.py:127`, `qg_model.py:141`, `demo_qg.py:508` (whose prose at L467 and L785 says the RHS "applies 2/3 dealiasing"), `pseudospectral_part2.py:381`, `scripts/*`, and `tests/test_correctness.py` (four tests).

## Proposed fix

Truncate the product in `advection_scalar` exactly as `jacobian` does (`return self.apply_dealias(u * dq_dx + v * dq_dy)`), in 2D and 3D. Also give `advection_scalar` the `spectral` argument the other operators have. If a non-dealiased variant is wanted for some reason, add an explicit `dealias: bool = True` argument rather than leaving the default aliased.

## Acceptance criteria

- [ ] Test: `jacobian(psi, q) == advection_scalar(*velocity_from_streamfunction(psi), q)` to 1e-10 on a band-limited random field at N = 48 (currently differs by 208).
- [ ] Test: energy of `advection_scalar(...)` outside the mask is < 1e-14 relative.
- [ ] 3D `advection_scalar` covered by the same test on a 24×16×32 grid.
- [ ] Notebook prose in `demo_qg.py` matches the behaviour.
